src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/services/GetRiskPredictorScoresForPersonServiceTest.kt

```diff
@@ -0,0 +1,144 @@
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.Identifiers
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.Person
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.Response
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.RiskPredictorScore
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.UpstreamApi
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.UpstreamApiError
+
+@ContextConfiguration(
+  initializers = [ConfigDataApplicationContextInitializer::class],
+  classes = [GetRiskPredictorScoresForPersonService::class],
+)
+internal class GetRiskPredictorScoresForPersonServiceTest(
+  @MockBean val assessRisksAndNeedsGateway: AssessRisksAndNeedsGateway,
+  @MockBean val getPersonService: GetPersonService,
+  private val getRiskPredictorScoresForPersonService: GetRiskPredictorScoresForPersonService,
+) : DescribeSpec(
+  {
+    val pncId = "1234/56789B"
+    val nomisNumber = "Z99999ZZ"
+    val crn = "X123456"
+
+    val personFromPrisonOffenderSearch =
+      Person(firstName = "Chandler", lastName = "Bing", identifiers = Identifiers(nomisNumber = nomisNumber))
+    val personFromProbationOffenderSearch =
+      Person(firstName = "Phoebe", lastName = "Buffay", identifiers = Identifiers(deliusCrn = crn))
+
+    beforeEach {
+      Mockito.reset(getPersonService)
+      Mockito.reset(assessRisksAndNeedsGateway)
+
+      whenever(getPersonService.execute(pncId = pncId)).thenReturn(
+        Response(
+          data = mapOf(
+            "prisonerOffenderSearch" to personFromPrisonOffenderSearch,
+            "probationOffenderSearch" to personFromProbationOffenderSearch,
+          ),
+        ),
+      )
+
+      whenever(assessRisksAndNeedsGateway.getRiskPredictorScoresForPerson(crn)).thenReturn(Response(data = emptyList()))
+    }
+
+    it("retrieves a person from getPersonService") {
+      getRiskPredictorScoresForPersonService.execute(pncId)
+
+      verify(getPersonService, VerificationModeFactory.times(1)).execute(pncId = pncId)
+    }
+
+    it("retrieves risk predictor scores for a person from ARN API using CRN") {
+      getRiskPredictorScoresForPersonService.execute(pncId)
+
+      verify(assessRisksAndNeedsGateway, VerificationModeFactory.times(1)).getRiskPredictorScoresForPerson(crn)
+    }
+
+    it("returns risk predictor scores for a person") {
+      val riskPredictors = listOf(
+        RiskPredictorScore(
+          generalPredictorScore = GeneralPredictorScore(80),
+        ),
+      )
+      whenever(assessRisksAndNeedsGateway.getRiskPredictorScoresForPerson(crn)).thenReturn(
+        Response(data = riskPredictors),
+      )
+
+      val response = getRiskPredictorScoresForPersonService.execute(pncId)
+
+      response.data.shouldBe(riskPredictors)
+    }
+
+    describe("when an upstream API returns an error") {
+
+      xdescribe("when a person cannot be found by pnc ID in probation offender search") {
```

Comment:
Are these tests skipped for a reason?

Reply:
Yup, the functionality for when someone can't be found isn't in the personService correctly yet. So these tests will be relevant for once that functionality has been added to the personService

Reply:
Cool beans, thanks for the clarification!